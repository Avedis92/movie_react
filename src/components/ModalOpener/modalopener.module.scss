@import "../sasscommonmixxins/flex.scss";
@import "../sasscommonmixxins/button.scss";
@import "../sasscommonmixxins/mediaqueries.scss";
.modalOpenerContainer {
  height: 100%;
  width: 100%;
  .infoItem {
    height: 100%;
    background: rgba(0, 0, 0, 0.705);
    background-size: cover;
    @include flexElements(flex, column, center, space-evenly);
    .watchNow {
      color: white;
      @include flexElements(flex, column, center, space-evenly);
      width: 70%;
      height: 40%;
      text-align: center;
      margin: auto;
      :nth-child(1) {
        cursor: pointer;
        transition: 500ms;
        &:hover {
          transform: scale(102%);
        }
      }
      p {
        font-size: 3vh;
      }
    }
    .viewInfoButton {
      @include styleButtons(
        40%,
        6vh,
        white,
        rgba(0, 0, 0, 0.63),
        none,
        1px solid white
      );
      margin: auto;
      transition: 500ms;
      cursor: pointer;
      &:hover {
        transform: scale(105%);
      }
    }
    @include mobile {
      width: 100%;
      height: 100%;
    }
  }
  @include mobile {
    width: 100%;
    height: 500px;
  }
}
