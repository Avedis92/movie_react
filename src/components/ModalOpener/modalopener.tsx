import { useState } from 'react';
import ModalVideo from 'react-modal-video';
import 'react-modal-video/scss/modal-video.scss';
import TrailerPlayer from '../TrailerPlayer/trailerplayer';
import styles from './modalopener.module.scss';

interface Modal {
  movieId:number,
  backgroundImage:string,
  handleViewInfoClick():void
}

function ModalOpener({ movieId, handleViewInfoClick, backgroundImage }:Modal) {
  const [isOpen, setIsOpen] = useState(false);
  const [trailerKey, setTrailerKey] = useState('');
  const {
    modalOpenerContainer, infoItem, watchNow, viewInfoButton,
  } = styles;
  return (
    <div className={modalOpenerContainer} style={{ backgroundImage: `url(${backgroundImage})` }}>
      {isOpen ? (
        <ModalVideo
          channel="youtube"
          isOpen={isOpen}
          videoId={trailerKey}
          onClose={() => setIsOpen(false)}
        />
      ) : null}
      <div className={infoItem}>
        <div className={watchNow}>
          <TrailerPlayer
            movieId={movieId}
            buttonType="playIcon"
            openMovieTrailers={(key) => {
              setTrailerKey(key);
              setIsOpen(true);
            }}
          />
          <p>Watch Now</p>
        </div>
        <button className={viewInfoButton} type="button" onClick={() => handleViewInfoClick()}>View Info</button>
      </div>
    </div>
  );
}
export default ModalOpener;
