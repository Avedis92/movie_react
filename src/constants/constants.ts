export const apiKey = '0600befd930f4be78f3d1123f393affc';
export const baseUrl = 'https://api.themoviedb.org/3/';
export const trendingMovieUrl = `${baseUrl}trending/movie/week?api_key=${apiKey}&page=`;
export const topRatedMovieUrl = `${baseUrl}movie/top_rated?api_key=${apiKey}&page=`;
export const upcomingMovieUrl = `${baseUrl}movie/upcoming?api_key=${apiKey}&page=`;
export const movieGenreListUrl = `${baseUrl}genre/movie/list?api_key=${apiKey}`;
export const searchMovieUrl = `${baseUrl}search/movie?api_key=${apiKey}&query=`;
export const postPathUrl = 'https://image.tmdb.org/t/p/w500';
export const defaultBackgroundImage = 'https://media-assets-05.thedrum.com/cache/images/thedrum-prod/s3-news-tmp-10557-netflixa--default--1280.jpg';
export const pageNotFoundBackgroundImage = 'https://cdn3.vectorstock.com/i/1000x1000/67/12/404-error-page-background-vector-20596712.jpg';
export const movieVideoUrl = `${baseUrl}/movie/`;
export const searchMovieByGenreUrl = `${baseUrl}discover/movie?api_key=${apiKey}&with_genres=`;
export const defaultPageNumber = 1;
export const defaultMovieType = 'trendings';
