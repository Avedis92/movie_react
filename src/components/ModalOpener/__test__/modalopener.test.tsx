import { render } from '@testing-library/react';
import ModalOpener from '../modalopener';
import { postPathUrl } from '../../../constants/constants';

test('Create a snapshot test', () => {
  const {asFragment} = render(<ModalOpener
    movieId={297761}
    handleViewInfoClick={() => null}
    backgroundImage={`${postPathUrl}/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg`}
  />);
  expect(asFragment()).toMatchSnapshot();
});
