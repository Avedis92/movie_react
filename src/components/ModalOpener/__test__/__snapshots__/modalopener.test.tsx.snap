// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="modalOpenerContainer"
    style="background-image: url(https://image.tmdb.org/t/p/w500/e1mjopzAS2KNsvpbpahQ1a6SkSn.jpg);"
  >
    <div
      class="infoItem"
    >
      <div
        class="watchNow"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(27, 186, 235); font-size: 10vh;"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
          />
        </svg>
        <p>
          Watch Now
        </p>
      </div>
      <button
        class="viewInfoButton"
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
</DocumentFragment>
`;
